qcodes/dataset/exporters/export_to_xarray.py

```diff
@@ -101,11 +102,26 @@ def load_to_xarray_dataset(dataset: DataSetProtocol, data: ParameterData) -> xr.
     xrdataset = xr.Dataset(
         cast(Dict[Hashable, xr.DataArray], data_xrdarray_dict))
 
-    for dim in xrdataset.dims:
-        if "index" != dim:
-            paramspec_dict = dataset.paramspecs[str(dim)]._to_dict()
-            xrdataset.coords[str(dim)].attrs.update(paramspec_dict.items())
-
+    _add_param_spec_to_xarray_coords(dataset, xrdataset)
     _add_metadata_to_xarray(dataset, xrdataset)
 
     return xrdataset
+
+
+def _add_param_spec_to_xarray_coords(
+    dataset: DataSetProtocol, xrdataset: Union[xr.Dataset, xr.DataArray]
+) -> None:
+    for coord in xrdataset.coords:
```

Comment:
It only cares about coordinates. so data_vars left to be updated here.
Shouldn't we have a similar loop to update those:

```
for var in xrdataset.data_vars:
    ....
```

Reply:
This currently happens inside _load_to_xarray_dataarray_dict_no_metadata thou a refactor of that would perhaps be nice